packages/react/src/auto/polaris/PolarisAutoTable.tsx

```diff
@@ -46,15 +59,51 @@ export const PolarisAutoTable = <
 ) => {
   const { onClick } = props;
 
-  const [{ rows, columns, metadata, fetching, error, page, search }, refresh] = useTable<GivenOptions, SchemaT, FinderFunction, Options>(
-    props.model,
-    {
-      select: props.select,
-      columns: props.columns,
-      pageSize: props.pageSize,
-      live: props.live,
-    } as any
+  const [{ rows, columns, metadata, fetching, error, page, search, sort }, refresh] = useTable<
+    GivenOptions,
+    SchemaT,
+    FinderFunction,
+    Options
+  >(props.model, {
+    select: props.select,
+    columns: props.columns,
+    pageSize: props.pageSize,
+    live: props.live,
+    sort: props.sort,
+    filter: props.filter,
+  } as any);
+
+  const [sortColumnIndex, setSortColumnIndex] = useState<number | undefined>(
+    columns ? getColumnIndex(columns, props.sort ? Object.keys(props.sort)[0] : undefined) : undefined
+  );
+  const [defaultColumnIndex, setDefaultColumnIndex] = useState(true);
+  const [sortDirection, setSortDirection] = useState<"ascending" | "descending" | undefined>(
+    getPresetSortDirection(props.sort ? Object.values(props.sort)[0] : undefined)
   );
+  const [disableSort, setDisableSort] = useState<boolean>(false);
+  const defaultSortDirection = getPresetSortDirection(props.sort ? Object.values(props.sort)[0] : "Descending");
+
+  useEffect(() => {
+    if (!sortColumnIndex && defaultColumnIndex) {
+      setSortColumnIndex(columns ? getColumnIndex(columns, props.sort ? Object.keys(props.sort)[0] : undefined) : undefined);
```

Comment:
I'm not sure I follow the need for this useEffect. when the columns change, we want to reset the sort back to the default?

Reply:
The `sortColumnIndex` state determines where the sort icon is: 
![image](https://github.com/user-attachments/assets/71cbaa02-244c-4f04-8473-8bef64be7317)
`columns` take some time to load, so when sortColumnIndex is initialized, `columns` is undefined; later on it is initialized, but sortColumnIndex was not being updated.